### PR TITLE
Update MongoDB tests

### PR DESCRIPTION
**Pull request by jsmrcka**
In `mongodb-panache-quickstart`:
- make use of QuarkusTestResource

In `mongodb-quickstart`:
- add tests
- use MongoDB 4.2.6 (same as `mongodb-panache-quickstart`)
- use QuarkusTestResource

Related guides update: https://github.com/quarkusio/quarkus/pull/20241.

**Check list**:

Your pull request:

- [x] targets the `development` branch
- [x] uses the `999-SNAPSHOT` version of Quarkus
- [x] has tests (`mvn clean test`)
- [x] works in native (`mvn clean package -Pnative`)
- [x] has native tests (`mvn clean verify -Pnative`)
- [ ] makes sure the associated guide must not be updated
- [x] links the guide update pull request (if needed)
- [ ] updates or creates the `README.md` file (with build and run instructions)
- [ ] for new quickstart, is located in the directory _component-quickstart_
- [ ] for new quickstart, is added to the root `pom.xml` and `README.md`

### CODE REVIEW

**Comment by rsvoboda** [comment]
According @jsmrcka failing checks is not related to the PR, @cescoffier can you pls review?

File: mongodb-quickstart/pom.xml [hunk 1]
```diff
@@ -40,6 +40,14 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-smallrye-context-propagation</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-resteasy-jackson</artifactId>
```
Comment (cescoffier) [inline mongodb-quickstart/pom.xml:45]: We should switch to RestEasy reactive (while keeping imperative code)

File: mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitService.java [hunk 2]
```diff
@@ -35,6 +35,6 @@ public void add(Fruit fruit) {
     }
 
     private MongoCollection<Fruit> getCollection() {
-        return mongoClient.getDatabase("fruit").getCollection("fruit", Fruit.class);
+        return mongoClient.getDatabase("fruit").getCollection("codec_fruit", Fruit.class);
```
Comment (cescoffier) [inline mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitService.java:38]: Why did you change the collection name?

File: mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitResource.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/codec_fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class CodecFruitResource {
```
Comment (cescoffier) [inline mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitResource.java:18]: I'm not sure about the codec prefix. Any reason for this?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitService.java (hunk 2 above)
Comment (jsmrcka) [reply mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitService.java:38]: To make it easier to run the same test on 3 different resources, each one of them using slightly different approach (classic, with codec, reactive).

File: mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitResource.java (hunk 3 above)
Comment (jsmrcka) [reply mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitResource.java:18]: I introduced this resource to complete the 3 approaches present in this QS, see also https://github.com/quarkusio/quarkus-quickstarts/pull/938#discussion_r711107489.

File: mongodb-quickstart/pom.xml (hunk 1 above)
Comment (jsmrcka) [reply mongodb-quickstart/pom.xml:45]: I missed this one, thanks for spotting.

File: mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitService.java (hunk 2 above)
Comment (cescoffier) [reply mongodb-quickstart/src/main/java/org/acme/mongodb/CodecFruitService.java:38]: Ah ok! Thanks for the clarification

**Comment by cescoffier** [comment]
Hum, while the issue looks unrelated, it seems to happen very early in the pipeline. It seems to be a gradle issue. @aloubyansky are you aware of any issue?

**Comment by aloubyansky** [comment]
````
2021-09-17T15:25:39.0219402Z [ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:exec (gradle) on project io.quarkus.extension.gradle.plugin: Command execution failed. Process exited with an error: 1 (Exit value: 1) -> [Help 1]
2021-09-17T15:25:39.0238720Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:exec (gradle) on project io.quarkus.extension.gradle.plugin: Command execution failed.
2021-09-17T15:25:39.0249775Z     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
````

The ` io.quarkus.extension.gradle.plugin` is a new Gradle plugin for extension development merged into main today.

````
2021-09-17T15:25:34.9751772Z FAILURE: Build failed with an exception.
2021-09-17T15:25:34.9752340Z 
2021-09-17T15:25:34.9752837Z * Where:
2021-09-17T15:25:34.9754623Z Settings file '/home/runner/work/quarkus-quickstarts/quarkus-quickstarts/quarkus/devtools/gradle/settings.gradle' line: 1
2021-09-17T15:25:34.9755674Z 
2021-09-17T15:25:34.9756186Z * What went wrong:
2021-09-17T15:25:34.9757672Z Could not compile settings file '/home/runner/work/quarkus-quickstarts/quarkus-quickstarts/quarkus/devtools/gradle/settings.gradle'.
2021-09-17T15:25:34.9758885Z > startup failed:
2021-09-17T15:25:34.9760538Z   settings file '/home/runner/work/quarkus-quickstarts/quarkus-quickstarts/quarkus/devtools/gradle/settings.gradle': 1: Only Project build scripts can contain plugins {} blocks
2021-09-17T15:25:34.9761829Z   
2021-09-17T15:25:34.9762883Z   See https://docs.gradle.org/5.6.3/userguide/plugins.html#sec:plugins_block for information on the plugins {} block
2021-09-17T15:25:34.9764691Z   
2021-09-17T15:25:34.9765238Z    @ line 1, column 1.
2021-09-17T15:25:34.9765784Z      plugins {
2021-09-17T15:25:34.9766290Z      ^
2021-09-17T15:25:34.9766749Z   
2021-09-17T15:25:34.9767219Z   1 error
2021-09-17T15:25:34.9767599Z 
2021-09-17T15:25:34.9767958Z 
2021-09-17T15:25:34.9768428Z * Try:
2021-09-17T15:25:34.9769651Z Run with --info or --debug option to get more log output. Run with --scan to get full insights.
2021-09-17T15:25:34.9770344Z 
2021-09-17T15:25:34.9770869Z * Exception is:
2021-09-17T15:25:34.9773141Z org.gradle.groovy.scripts.ScriptCompilationException: Could not compile settings file '/home/runner/work/quarkus-quickstarts/quarkus-quickstarts/quarkus/devtools/gradle/settings.gradle'.
````

@glefloch does it tell you anything?

**Comment by glefloch** [comment]
This is really weird, the entreprise plug-in was also applied there before the new plugin introduction. The issue come from the version of the gradle wrapper used in the project (5.6.3). I will try push an update ASAP.

**Comment by cescoffier** [comment]
I just merged the gradle wrapper update which fixes the issue. @jsmrcka  Can you rebase this PR?

**Comment by jsmrcka** [comment]
> I just merged the gradle wrapper update which fixes the issue. @jsmrcka Can you rebase this PR?

The update is in `main`. Should I change the base of this PR from `development` to `main`?

**Comment by cescoffier** [comment]
No, you still need to target the `development` branch.

**Comment by jsmrcka** [comment]
@cescoffier But the fix is not present in current `development`. So, do you want me to rebase the whole `development` branch along with the branch of mine?

**Comment by cescoffier** [comment]
@glefloch can you also fix the development branch?

**Comment by cescoffier** [comment]
Thanks for the update @jsmrcka - waiting for the CI to complete.

**Comment by cescoffier** [comment]
Thanks again @jsmrcka!

**Comment by jsmrcka** [comment]
@cescoffier Thank you!
